subsystems/photonVisionSubsystem.py

```diff
@@ -0,0 +1,152 @@
+        
+        # Start camera server 
+        self.camera = PhotonCamera("Limelight")
+        
+        # gets latest result from camera
+        self.result = self.camera.getLatestResult()
+        
+        # Checks if limelight detects targets
+        self.hasTargets = self.result.hasTargets()
+        
+        # Recieves the latest targets seen by the limelight
+        self.targets = self.result.getTargets()
+        
+        # April Tag Layout   (Currently empty/ not being used)
+        self.aprilTagLayout = photonPoseEstimator.AprilTagFieldLayout()
+        
+        # # Position estimation strategy that is used by the PhotonPoseEstimator class
+        self.PoseStrat = photonPoseEstimator.PoseStrategy.CLOSEST_TO_REFERENCE_POSE
+        
+        
+    def isDetecting(self, id: int) -> bool: # Check if desired april tag is beeing seen
+      for i in self.targets:
+         if i.getFiducialId() == id:
+            return True
+      return False
+
+    def getTagOdometry(self, id: int): # Get the yaw, pitch, and skew value of desired target if seen  
+      for i in self.targets:
+         if i.getFiducialId() == id:
+            return [i.getYaw(), i.getPitch(), i.getSkew()]
+        
+    def togglePipeLine(self):
+        # Change PipleLine to desired one
+        if (self.camera.getPipelineIndex() == 0):
+           self.camera.setPipelineIndex(1)
+        else:
+           self.camera.setPipelineIndex(0)
+            
+    def captureImage(self):
+        # Capture pre-process camera stream image
+        self.camera.takeInputSnapshot()
+
+        # Capture post-process camera stream image
+        self.camera.takeOutputSnapshot()
+        
+    def nothingCommand(self):
+        return False
+    
+    
+    # Next 10 methods will return info about the target beeing seen
+         
+    def getTargetYaw(self, id: int) -> float: # Return the yaw of the desired target if seen
+      for i in self.targets:
+         if i.getFiducialId() == id:
+            return i.getYaw()
+          
+    def getTargetPitch(self, id: int) -> float: # Return the yaw of the desired target if seen
+      for i in self.targets:
+         if i.getFiducialId() == id:
+            return i.getPitch()
+          
+    def getTargetSkew(self, id: int) -> float: # Return the Skew of the desired target if seen
+      for i in self.targets:
+         if i.getFiducialId() == id:
+            return i.getSkew()
+
+    def GetTargetArea(self, id: int) -> float: # returns the percentage of area the target takes up on the camera
+      for i in self.targets:
+         if i.getFiducialId() == id:
+            return i.getArea()
+      return 0
+          
+    def getTargetID(self) -> tuple: # Return a list of tags beeing seen
+      listId = ()
+      for i in self.targets:
+         listId += i.getFiducialId()
+      return listId
+    
+    def getTargetDistance(self, id: int): # Returns the X distance of the Desired ID Tag if seeen
```

Comment:
What are you expecting these methods to return if `id` is not in `self.targets`?